### PR TITLE
feat: = + > then ⇒

### PR DESCRIPTION
**Pull request by Gen1us1100**
Fixes #2309

offers users to use '=' + '>' shortcut to input '⇒' into the editor.
Below video showcases the proper functioning 

https://github.com/AppFlowy-IO/appflowy-editor/assets/85549793/79ce67c9-b6d7-4742-892d-146178fd03a5

### CODE REVIEW

**Comment by LucasXu0** [comment]
@Gen1us1100 Please add tests to cover this functionality. You can refer to the `format_bold_test.dart`.

File: lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/format_double_characters/format_equal_greater_character.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  key: 'format = + > into ⇒',
+  character: _greater,
+  handler: (editorState) async => handleEqualGreaterReplacement(
+    editorState: editorState,
+    character: _greater,
+    replacement: _arrow,
+  ),
+);
+
+Future<bool> handleEqualGreaterReplacement({
```
Comment (Xazin) [inline lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/format_double_characters/format_equal_greater_character.dart:24]:
```suggestion
// TODO(Xazin): Combine two character replacement methods into
//  a helper function
Future<bool> handleEqualGreaterReplacement({
```

**Comment by Gen1us1100** [comment]
hey, @LucasXu0 I have tried adding tests for the stated feature, do check it out!

File: lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/format_double_characters/format_equal_greater_character.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import 'package:appflowy_editor/appflowy_editor.dart';
+
+const _greater = '>';
+const _equals = '=';
+const _arrow = '⇒'; 
+
+// /// format '=' + '>' into an ⇒ 
+// ///
```
Comment (MayurSMahajan) [inline lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/format_double_characters/format_equal_greater_character.dart:8]: No need to commend the documentation comments.

Comment (MayurSMahajan) [reply lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/format_double_characters/format_equal_greater_character.dart:8]: So I mean that this: 

```dart
// /// format '=' + '>' into an ⇒ 
// ///
// /// - support
// ///   - desktop
// ///   - mobile
// ///   - web
// ///
```

should be:

```dart
/// format '=' + '>' into an ⇒ 
///
/// - support
///   - desktop
///   - mobile
///   - web
///
```

File: test/new/service/shortcuts/character_shortcut_events/double_characters_shortcut_events/format_equal_greater_test.dart [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      await editor.ime.typeText(_equals);
+      await editor.ime.typeText(_greater);
+
+      final delta = editor.nodeAtPath([0])!.delta!;
+      expect(delta.length, 1);
+      expect(delta.toPlainText(), _arrow);
+
+      await editor.dispose();
+    });
+    });
```
Comment (MayurSMahajan) [inline test/new/service/shortcuts/character_shortcut_events/double_characters_shortcut_events/format_equal_greater_test.dart:26]: great job! add more tests like:

1. executing this in a non-empty paragraph.
2. try doing undo and redo after inserting this.

**Review by MayurSMahajan — CHANGES_REQUESTED** [review]
The file location is indeed correct. Nice clean code.

**Pull request merged**